pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -315,7 +315,7 @@ func (c *InstallerController) manageInstallationPods(ctx context.Context, operat
 			// if we make a change to this status, we want to write it out to the API before we commence work on the next node.
 			// it's an extra write/read, but it makes the state debuggable from outside this process
 			if !equality.Semantic.DeepEqual(newCurrNodeState, currNodeState) {
-				klog.Infof("%q moving to %v because %s", currNodeState.NodeName, spew.Sdump(*newCurrNodeState), reason)
+				klog.Infof("%q moving to %v because %s and of expected revision %d", currNodeState.NodeName, spew.Sdump(*newCurrNodeState), reason, newCurrNodeState.CurrentRevision)
```

Comment:
as discussed in chat, `spew.Sdump(*newCurrNodeState)` includes `newCurrNodeState.CurrentRevision`. I would prefer to see a case distinction here that gets rid of the generic `spew.Sdump(*newCurrNodeState)` and prints proper human readible errors instead. One of them is what you have in mind with this change: `foo is moving to %v because pod bar is ready and it has the target revision 42`.

Reply:
OK I've tried to find a solution. This is the test output: `I0930 21:49:41.820918   18477 installer_controller.go:316] "test-node-1" is moving to "test-node-1" with Target Revision: 0 and Current Revision: 1 because static pod is ready and reached the Expected Revision 1
`